File: src/main/java/gestorAplicacion/servicios/Tarjeta.java
package gestorAplicacion.servicios;

import gestorAplicacion.instalaciones.Instalacion;

import java.io.Serializable;
import java.util.ArrayList;

public class Tarjeta implements Serializable {
	/*La clase tarjeta solo maneja temas de informacion que
	* van a ser utilizados para las restricciones de acceso a instalaiones*/
	
	//private static final long serialVersionUID = 1L;
	/*Constantes*/
	private int atraccionesMontadas = 0;
	private final String[] tipoTarjeta = {"Adulto", "Infante"};
	private final float[] costoTarjeta = {100, 50};
	/*Atributos*/
	protected int idTarjeta;
        private float costo;
	private int tipoT;
	private boolean tarjetaFisica;
	private boolean activa = false; //Por defecto la tarjeta no esta activa hasta que se realice el pago de la misma
        private int cantidadDeEntradas = 0;
        private float saldo = 0;

		public Tarjeta() {}
	public Tarjeta(int idTarjeta, int tipoTarjeta) {
		//this.reserva = reserva;
		this.idTarjeta = idTarjeta;
                this.tipoT = tipoTarjeta;      
                
        }

	public boolean cargarTarjeta(Tarjeta tarjeta, float saldo){     //Para agregar saldo a una tarjeta se verifica qeu este activa
		if(tarjeta.getActiva()){
			tarjeta.agregarSaldo(saldo);
			return true;
		}
		else{
			return false;
		}
	}
                
        public void agregarEntrada(){
            this.cantidadDeEntradas ++;
            
            if (cantidadDeEntradas == 3){
                System.out.println("Aplica para descuento") ;
                this.cantidadDeEntradas=0;
            }
            else{ ;
                System.out.println("Faltan "+ (3-cantidadDeEntradas) +" entradas para tener descuento");
            }
        }
		public void setSaldo(float saldo) {
			this.saldo = saldo;
		}
                
        public void agregarSaldo(float saldo){
            this.saldo += saldo;
        }
                               
	
	public int getIdTarjeta() {
		return idTarjeta;
	}
	public void setIdTarjeta(int idTarjeta) {
		this.idTarjeta = idTarjeta;
	}
	public void asignarCosto() {
		this.saldo = costoTarjeta[tipoT];
	}
	public float getSaldo() {
		return saldo;
	}
	public String getTipoTarjeta() {
		return tipoTarjeta[tipoT];
	}
	public boolean getActiva(){
		return activa;
	}
	public void setActiva(boolean activa){
		this.activa = activa;
	}
	public boolean getTarjetaFisica(){
		return tarjetaFisica;
	}
	public void setTarjetaFisica(boolean tarjetaFisica){
		this.tarjetaFisica = tarjetaFisica;
	}

    @Override
    public String toString() {
        return "Tarjeta{" + "tipoTarjeta=" + tipoTarjeta[tipoT] + ", costoTarjeta=" + costoTarjeta[tipoT] + ", idTarjeta=" + idTarjeta + ", costo=" + costo +", tarjetaFisica= " + tarjetaFisica + ", activa= " + activa + ", cantidadDeEntradas= " + cantidadDeEntradas + ", saldo= " + saldo + '}';
    }
	public void sumarAtracciones() {
		this.atraccionesMontadas++;
	}
	public int getAtracciones() {
		return this.atraccionesMontadas;
	}


     
        
}

File: practica2/src/gestorAplicacion/servicios/tarjeta.py

class Tarjeta:

    tipoTarjeta = {"Adulto", "Infante"}
    costoTarjeta = {100, 50}
    atraccionesMontadas = 0
    def __init__(self, idTarjeta, costo, tipoT,):
        self._idTarjeta = idTarjeta
        self._costo = costo 
        self._tipoT = tipoT
        self._tarjetaFisica = False
        self._activa = False
        self._cantidadDeEntradas = 0
        self._saldo = 0
        self._atraccionesMontadas = 0


    def __str__(self):
        return f"Id: {self.getIdTarjeta()}, Tipo: {self._tipoT}, Entradas: {self._cantidadDeEntradas}, Saldo: {self._saldo}, Estado: {self._activa}"

    def getTipo(self):
        return self._tipoT

    def getIdTarjeta(self):
        return self._idTarjeta

    def getActiva(self):
        return self._activa

    def getSaldo(self):
        return self._saldo

    def setSaldo(self, saldo):
        self._saldo = saldo

    def cargarTarjeta(self, tarjeta, saldo):
        tarjeta.agregarSaldo(saldo)
        return True

    def agregarSaldo(self, saldo):
        self._saldo += saldo

    
    def sumarAtracciones(self):
        self._atraccionesMontadas += 1
        if self.getAtracciones() ==6 :              #Si ha montado en 5 atracciones vuelve a empezar para el descuento
            self._atraccionesMontadas = 0

    def getAtracciones(self):
        return self._atraccionesMontadas
        


    def agregarEntrada(self):
        self._cantidadDeEntradas +=1
        if self._cantidadDeEntradas == 3:
            self._cantidadDeEntradas=0
            print ("\n**Aplica para descuento**\n")
            return True
        else:
            return print("Faltan ",  3-self._cantidadDeEntradas ," visitas al parque para tener descuento")

    def setTarjetaFisica(self):
        self._tarjetaFisica = True

    def getTarjetaFisica(self):
        return self._tarjetaFisica

    def activarTarjeta(self):
	    self._activa = True
